src/ec/eddsa.rs

```diff
@@ -96,9 +96,7 @@ impl<'a> Ed25519KeyPair {
 
     fn from_bytes_unchecked(private_key: &[u8], public_key: &[u8])
                             -> Result<Ed25519KeyPair, error::Unspecified> {
-        if private_key.len() != 32 {
-            return Err(error::Unspecified);
-        } else if public_key.len() != 32 {
+        if private_key.len() != 32 || public_key.len() != 32 {
```

Comment:
I still prefer to keep this as two if statements. Let's replace "} else " with whitespace to move the "if public_key_len() != 32" to the next line.

Separate statements help with code coverage.